app/controllers/users_controller.rb

```diff
@@ -1,4 +1,13 @@
 class UsersController < ApplicationController
+  # Before filters
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :admin_user, only: :destroy
```

Comment:
Question for research: What do you think about pros/cons of before_filter solution? How we can replace it?